shapeshift/web
# feat: wallet identifiers

## Description

implements #728

## Notice

Before submitting a pull request, please make sure you have answered the following:


## Pull Request Type


## Issue (if applicable)

closes #728

## Testing

Please outline all testing steps

1. Setup local dev environment
2. Connect a wallet
3. Notice the new identifier in the top right

## Screenshots (if applicable)
Dark mode:
<img width="236" alt="Screen Shot 2022-01-29 at 4 18 47 AM" src="https://user-images.githubusercontent.com/35400192/151641581-9d05ef02-b46f-4d99-9d35-ef6c6f7c19a1.png">
Light Mode:
<img width="234" alt="Screen Shot 2022-01-29 at 4 19 07 AM" src="https://user-images.githubusercontent.com/35400192/151641590-c2b46c36-f56c-48b4-a8c7-ba27392c25ca.png">

## Additional Info
As this is supposed to be a bounty I am attaching the time I spent working on this: 5 hours

> i took approval from @0xean before starting work on this bounty

~~**NOTE:** While working on this I noticed that when switching between wallets multiple times an issue emerges, the wallet identifier does not update. I tracked this issue down to a certain dependency called `react-middle-ellipsis` and confirmed that when the above mentioned dependency is not used this issue is solved. I'm not exactly sure why this dependency causes this behavior and did not investigate too deep into it as it was out of scope of this bounty. I'm leaving this note here as this will need to be fixed later on.~~

I solved this by making a custom implementation of the ellipsis system
